Utilities/StaticAnalyzers/src/ClassChecker.cpp

```diff
@@ -650,35 +630,32 @@ void ClassChecker::checkASTDecl(const clang::CXXRecordDecl *RD, clang::ento::Ana
                     clang::ento::BugReporter &BR) const {
 
      const clang::SourceManager &SM = BR.getSourceManager();
-     const char *sfile=SM.getPresumedLoc(RD->getLocation()).getFilename();
+     const char *sfile=SM.getPresumedLoc(RD->getLocation()).getFilename();     
      if (!support::isCmsLocalFile(sfile)) return;
-     
      std::string buf;
      llvm::raw_string_ostream os(buf);
      std::string name = RD->getQualifiedNameAsString();
      if ( ! support::isDataClass(name) ) return;
-
      for ( auto I = RD->field_begin(), E = RD->field_end(); I != E; ++I)
           {
           const FieldDecl * D = (*I) ;
-          if ( D->hasAttr<CMSThreadGuardAttr>() || D->hasAttr<CMSThreadSafeAttr>()) return;
-          if ( D->isMutable() )
+	   if ( D->hasAttr<CMSThreadGuardAttr>() || D->hasAttr<CMSThreadSafeAttr>()) return;
+           if ( D->isMutable() )
                 {
                     clang::QualType t =  D->getType();
                     clang::ento::PathDiagnosticLocation DLoc =
                     clang::ento::PathDiagnosticLocation::createBegin(D, BR.getSourceManager());
                     if ( support::isSafeClassName( t.getCanonicalType().getAsString() ) ) return;
-                    if ( ! support::isDataClass( support::getQualifiedName(*RD) ) ) return;
-                    WalkAST walker(this,BR, mgr.getAnalysisDeclContext(RD), (*(RD->ctor_begin()))->getMostRecentDecl() ) ;
```

Comment:
This caused a crash because RD->ctor_begin() when returned a null pointer. Making fixAnonNS a helper function removed the need to instantiate a WalkAST to get access to fixAnonNS.